File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/SearchResultHolder.java
package org.fossasia.susi.ai.chat.adapters.viewholders;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import org.fossasia.susi.ai.R;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * <h1>Search result view holder</h1>
 *
 * Created by saurabh on 19/11/16.
 */
public class SearchResultHolder extends RecyclerView.ViewHolder {

    @BindView(R.id.link_preview_image)
    public ImageView previewImageView;
    @BindView(R.id.link_preview_title)
    public TextView titleTextView;
    @BindView(R.id.link_preview_description)
    public TextView descriptionTextView;
    @BindView(R.id.preview_layout)
    public LinearLayout previewLayout;
    @BindView(R.id.link_preview_text_layout)
    public LinearLayout previewTextLayout;

    /**
     * Instantiates a new Search result holder.
     *
     * @param itemView the item view
     */
    public SearchResultHolder(View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
    }
}

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/SearchResultHolder.kt
package org.fossasia.susi.ai.chat.adapters.viewholders

import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import kotterknife.bindView
import org.fossasia.susi.ai.R

class SearchResultHolder(itemView: View) : androidx.recyclerview.widget.RecyclerView.ViewHolder(itemView) {

    val previewImageView: ImageView by bindView(R.id.link_preview_image)
    val titleTextView: TextView by bindView(R.id.link_preview_title)
    val descriptionTextView: TextView by bindView(R.id.link_preview_description)
    val previewLayout: LinearLayout by bindView(R.id.preview_layout)
    val previewTextLayout: LinearLayout by bindView(R.id.link_preview_text_layout)
}
